dataeng/resources/warehouse-transforms-master-ci.sh

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env bash
+set -ex
+
+# Setup to run python script to create snowflake schema
+cd $WORKSPACE/analytics-tools/snowflake
+make requirements
+
+# Download Prod build manifest.json file from S3 and creating directory to place manifest file.
+cd $WORKSPACE/ && mkdir -p manifest
+
+pip install awscli
+
+aws s3 cp s3://edx-dbt-docs/manifest.json ${WORKSPACE}/manifest
+
+# Setup to run dbt commands
+cd $WORKSPACE/warehouse-transforms
+
+PREV_COMMIT_ID=$(git rev-parse @~)
+
+git diff $PREV_COMMIT_ID --name-only
+
+# Finding the project names which has changed in this PR. Using git diff latest to compare this branch from master
+# It returns all the files name with full path. Searching through it using egrep to find which project(s) the changing files belong.
+# It might happen one PR may be changing files in different projects.
+if git diff $PREV_COMMIT_ID --name-only | egrep "projects/reporting" -q; then isReporting="true"; else isReporting="false"; fi
+if git diff $PREV_COMMIT_ID --name-only | egrep "projects/automated/applications" -q; then isApplications="true"; else isApplications="false"; fi
+if git diff $PREV_COMMIT_ID --name-only | egrep "projects/automated/raw_to_source" -q; then isRawToSource="true"; else isRawToSource="false"; fi
+if git diff $PREV_COMMIT_ID --name-only | egrep "projects/automated/telemetry" -q; then isTelemetry="true"; else isTelemetry="false"; fi
+
+
+# Setup to run dbt commands
+cd $WORKSPACE/warehouse-transforms
+
+# To install right version of dbt
+pip install -r requirements.txt
+
+if [ "$isReporting" == "true" ]
+then
+
+    cd $WORKSPACE/analytics-tools/snowflake
+
+    # Schema_Name will be the Github Pull Request ID e.g. 1724 prefixed with 'PR_*' and sufixed with project name e.g. PR_1724_reporting
+    export CI_SCHEMA_NAME=PR_${ghprbPullId}_reporting_merged
+    # Schema is dynamically created against each PR. It is the PR number with 'PR_*' as prefixed.
+    # profiles.yml contains the name of Schema which is used to create output models when dbt runs. 
+    python create_ci_schema.py --key_path $KEY_PATH --passphrase_path $PASSPHRASE_PATH --automation_user $USER --account $ACCOUNT --db_name $DB_NAME --schema_name $CI_SCHEMA_NAME 
+    
+    DBT_PROJECT_PATH='reporting'
+    # This is a Slim CI syntax used to "run" only modified and downstream models
+    DBT_RUN_OPTIONS="-m state:modified+ @state:modified --defer --state $WORKSPACE/manifest" 
```

Comment:
excluded `1+test_type:data` from models creation tag as it was suggested to use in v0.18.1 as an alternative to not being able to use --defer state in tests, we are now on v0.19.1 so removing it.